tool/tsh/common/app_azure.go

```diff
@@ -91,12 +105,37 @@ func newAzureApp(tc *client.TeleportClient, cf *CLIConf, appInfo *appInfo) (*azu
 	return &azureApp{
 		localProxyApp: localProxyApp,
 		cf:            cf,
-		msiSecret:     msiSecret,
+		tokenSecret:   msiSecret,
+		fetchCLIVersion: sync.OnceValues(func() (*semver.Version, error) {
+			// Retrieve the core version as it contains the login-related changes.
+			versionInfo := struct {
+				CLICoreVersion string `json:"azure-cli-core"`
+			}{}
+
+			var buf bytes.Buffer
+			cmd := exec.Command(azureCLIBinaryName, "version")
+			cmd.Stdout = &buf
+			if err := cf.RunCommand(cmd); err != nil {
+				return nil, trace.Wrap(err)
+			}
+
+			if err := json.Unmarshal(buf.Bytes(), &versionInfo); err != nil {
+				return nil, trace.Wrap(err)
+			}
+
+			ver, err := semver.NewVersion(versionInfo.CLICoreVersion)
+			return ver, trace.Wrap(err)
+		}),
 	}, nil
 }
 
-// getMSISecret will try to find the secret by parsing MSI_ENDPOINT env variable if present; it will return random hex string otherwise.
-func getMSISecret() (string, error) {
+// getAzureTokenSecret will try to find the secret from the environment.
+// If not found it will return random hex string.
+func getAzureTokenSecret() (string, error) {
+	if identityHeader := os.Getenv("IDENTITY_HEADER"); identityHeader != "" {
+		return identityHeader, nil
+	}
+
 	endpoint := os.Getenv("MSI_ENDPOINT")
 	if endpoint == "" {
```

Comment:
Should we look for `IDENTITY_ENDPOINT` too? I know we will overwrite it, but perhaps checking its existence might be worth it for debugging purposes?